beluga_ros/include/beluga_ros/point_cloud.hpp

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_ROS_POINT_CLOUD_HPP
+#define BELUGA_ROS_POINT_CLOUD_HPP
+
+#include <range/v3/view/iota.hpp>
+
+#include <beluga/sensor/data/point_cloud.hpp>
+#include <beluga/views/take_evenly.hpp>
+#include <beluga_ros/messages.hpp>
+
+#include <sophus/se3.hpp>
+
+/**
+ * \file
+ * \brief Implementation of `sensor_msgs/PointCloud2` wrapper type.
+ */
+
+namespace beluga_ros {
+
+/// Thin wrapper type for 3D `sensor_msgs/PointCloud2` messages.
+class PointCloud2 : public beluga::BasePointCloud<PointCloud2> {
```

Comment:
@pvela2017 hmm, why `beluga_ros::PointCloud2`? `beluga_ros::PointCloud3d` would be at least consistent with `Eigen` nomenclature.

Reply:
I used PointCloud2 because the wrapper is for a PointCloud2 message.

Reply:
Let's use a name that is semantically meaningful rather than carrying forward with ROS naming conventions. Particularly for `PointCloud2`, which is called `PointCloud2` simply because `PointCloud` was already taken at the time it was introduced.